esmvalcore/preprocessor/_volume.py

```diff
@@ -603,3 +605,80 @@ def extract_trajectory(
     points = [("latitude", latitudes), ("longitude", longitudes)]
     interpolated_cube = interpolate(cube, points)  # Very slow!
     return interpolated_cube
+
+
+def _get_first_unmasked_data(array: da.Array, axis: int) -> da.Array:
+    """Get first unmasked value of an array along an axis."""
+    mask = da.ma.getmaskarray(array)
+    numerical_mask = da.where(mask, -1.0, 1.0)
+    indices_first_positive = da.argmax(numerical_mask, axis=axis)
+    indices = da.meshgrid(
+        *[da.arange(array.shape[i]) for i in range(array.ndim) if i != axis],
+        indexing="ij",
+    )
+    indices = list(indices)
+    indices.insert(axis, indices_first_positive)
+    first_unmasked_data = np.array(array)[tuple(indices)]
+    return first_unmasked_data
```

Comment:
```suggestion
def _get_first_unmasked_data(
    array: np.ndarray | da.Array,
    axis: int,
) -> np.ndarray | da.Array:
    """Get first unmasked value of an array along an axis."""
    npx = get_array_module(array)
    mask = npx.ma.getmaskarray(array)
    numerical_mask = npx.where(mask, -1.0, 1.0)
    indices_first_positive = npx.argmax(numerical_mask, axis=axis)
    indices = npx.meshgrid(
        *[npx.arange(array.shape[i]) for i in range(array.ndim) if i != axis],
        indexing="ij",
    )
    indices = list(indices)
    indices.insert(axis, indices_first_positive)
    first_unmasked_data = npx.array(array)[tuple(indices)]
    return first_unmasked_data
```

To make it work with numpy and dask arrays. This needs `from esmvalcore.preprocessor._shared import get_array_module`.